include/veriblock/blockchain/chain.hpp

```diff
@@ -98,8 +98,8 @@ struct Chain {
   void disconnectTip() { chain.pop_back(); }
 
   friend bool operator==(const Chain& a, const Chain& b) {
-    // TODO: think how to use use startHeight_
-    return a.chain.size() == b.chain.size() && a.tip() == b.tip();
+    // sizes may vary, so compare tips
+    return a.tip() == b.tip();
```

Comment:
IMO equal chains should contain equal elements.

Reply:
if tips of chains are equal, then chains are equal. In original BTC code there's also comparison for chain sizes... here, it is possible that chain size will be different, because they are stored from "startHeight", not from 0